src/storage/mvcc/txn.rs

```diff
@@ -161,16 +161,17 @@ impl<'a> MvccTxn<'a> {
             }
             _ => {
                 return match try!(self.reader.get_txn_commit_ts(key, self.start_ts)) {
-                    // Committed by concurrent transaction.
-                    Some(_) => Ok(()),
-                    // Rollbacked by concurrent transaction.
+                    Some((_, WriteType::Rollback)) |
                     None => {
+                        // Rollbacked by concurrent transaction.
```

Comment:
When commit a key that is not locked, it must be committed or rolled back. So we should panic for _None_.